playbook/controllers.go

```diff
@@ -0,0 +1,89 @@
+package playbook
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+type RunResponse struct {
+	RunID     string `json:"run_id"`
+	CreatedAt string `json:"created_at"`
+}
+
+type RunParams struct {
+	ID          uuid.UUID         `json:"id"`
+	ConfigID    uuid.UUID         `json:"config_id"`
+	ComponentID uuid.UUID         `json:"component_id"`
+	Params      map[string]string `json:"params"`
+}
+
+func (r *RunParams) Valid() error {
+	if r.ID == uuid.Nil {
+		return fmt.Errorf("playbook id is required")
+	}
+
+	if r.ConfigID == uuid.Nil && r.ComponentID == uuid.Nil {
+		return fmt.Errorf("either config_id or component_id is required")
+	}
+
+	if r.ConfigID != uuid.Nil && r.ComponentID != uuid.Nil {
+		return fmt.Errorf("either config_id or component_id is required")
+	}
+
+	return nil
+}
+
+// HandlePlaybookRun handles playbook run requests.
+func HandlePlaybookRun(c echo.Context) error {
```

Comment:
I think this should queue the playbook run, with a startdate of now and then we have a background process that runs them